main.go

```diff
@@ -64,15 +71,25 @@ func main() {
 }
 
 func run(c *cli.Context) {
-
 	if c.Bool("debug") {
 		logrus.SetLevel(logrus.DebugLevel)
 		logrus.SetReportCaller(true)
 	}
 
-	logrus.Info("Starting Terraform Controller")
-	kubeconfig, err := resolvehome.Resolve(c.String("kubeconfig"))
+	ctx := signals.SetupSignalHandler(context.Background())
 
+	go startController(ctx, c)
+	go startApi(ctx, c)
```

Comment:
I think doing the flag check here would be cleaner, also makes the intention clear as to what we are running vs having to check the function call to see the flag check

Reply:
sounds good